pkg/storage/disk.go

```diff
@@ -145,10 +146,35 @@ func (s *Module) DiskWrite(name string, image string) error {
 		return fmt.Errorf("image size is bigger than disk")
 	}
 
-	_, err = io.Copy(file, source)
-	if err != nil {
+	// writing the image concurrently to speedup the previous sequential write.
+	// the sequential write is slow because the data source is from the remote server.
+	var (
+		// use errgroup because there is no point in continuing if one of the goroutines failed
+		group               = new(errgroup.Group)
+		concurrentNum int   = 5
+		imgSize       int64 = imgStat.Size()
+		chunkSize           = imgSize / int64(concurrentNum)
+	)
+
+	log.Info().Int("concurrentNum", concurrentNum).Msg("writing image concurrently")
+	for i := 0; i < concurrentNum; i++ {
+		index := i
+		group.Go(func() error {
+			start := chunkSize * int64(index)
+			len := chunkSize
+			if index == concurrentNum-1 { //last chunk
+				len = imgSize - start
+			}
+			wr := io.NewOffsetWriter(file, start)
+			rd := io.NewSectionReader(source, start, len)
+			_, err = io.Copy(wr, rd)
```

Comment:
I believe using io.CopyBuffer could avoid temporal buffers https://pkg.go.dev/io#CopyBuffer

Reply:
also, no need to call file.Sync?

Reply:
> I believe using io.CopyBuffer could avoid temporal buffers https://pkg.go.dev/io#CopyBuffer

the difference between `io.Copy` and `CopyBuffer` is : CopyBuffer lets us choose our own buffer.
`io.Copy` will also use buffer.
The slowness here is because of downloading the file from the `hub`, so i don't think that the buffer will make big differences here.

Reply:
> also, no need to call file.Sync?

i don't think so, we're copying to `rfs` anyway.
CMIIW

Reply:
io.CopyBuffer uses a buffer your provide instead of io.Copy keeping allocating them during every call (it's still an io to save imo)